### DISCUSSION TITLE
no energy source available to power the inverter from

### DISCUSSION THREAD

**Discussion by Michel83** [hoylabs/OpenDTU-OnBattery#985] [Q&A]
Hallo,

kann mir jemand sagen was dieser Fehler ist ?

Die Reglung selbst Funktioniert aber der Wechselrichter wird nicht mehr eingeschaltet. 

Ich hatte lediglich den Wechselrichter gegen einen HMS-1000 gewechselt, der HMS-450 Funktionierte ohne diese Problem.

Wenn ich den Wechselrichter über OpenDTU neu Starte geht es wieder eine Zeitlang.

```
19:54:38.744 > [DPL::loop] ******************* ENTER **********************
19:54:38.801 > [DPL::loop] battery interface enabled, SoC: 65 %, StartTH: 85 %, StopTH: 21 %, SoC age: 0 s, ignore: no
19:54:38.857 > [DPL::getBatteryVoltage] BMS: 26.61 V, MPPT: 26.60 V, inverter: 26.60 V, returning: 26.61V
19:54:38.916 > [DPL::loop] dcVoltage: 26.61 V, loadCorrectedVoltage: 26.61 V, StartTH: 28.40 V, StopTH: 25.40 V
19:54:39.018 > [DPL::loop] StartTH reached: no, StopTH reached: no, SolarPT enabled, use at night: no
19:54:39.069 > [DPL::calcPowerLimit] battery use prevented, solar power (DC): 0 W
19:54:39.130 > [DPL::announceStatus] no energy source available to power the inverter from
19:54:39.192 > [DPL::announceStatus] the system is stable, the last power limit is still valid
19:54:39.246 > Success
```

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/36561525/3e58f906-5d04-436f-a6c0-c70a9dd88bf2)

Vielleicht hat ja jemand eine Idee wäre toll, Grüße.

**Comment by swingstate** [comment]
Deine Startspannung liegt bei 28,4V. Dein System hat aktuell mit 26,6V eine zu geringe Spannung. SOC ist auch unterhalb des Startwerts.

**Comment by Michel83** [comment]
Das ist nachdem der Wechselrichter ausgegangen ist. 

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/36561525/71ef0500-f14d-448a-bc7b-efe9de7d713e)

hier sieht man es ganz gut das er abgeschaltet hat und ein Neustart bei 20:00

**Comment by Michel83** [comment]
Sorry noch vergessen : 
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/36561525/e8240f3b-8f68-4396-a822-00ce03d0f32b)

**Reply by Manos1966** [reply]
> no energy source available to power the inverter from 19:54:39

- Was ist dein Setup?
- Was fuer eine Batterie? 
  - CANBUS Anschluss?
  - Volts? 
  - Wie kann es sein, dass erst mit 28,4V sie anfangen soll Strom zu liefern? (ich kenne 12V, 24V, 25V, 48V Batterien..aber 28V????

Funktioniert deine OpenDTU ohne Neustarts?
- Stelle Batterienutzung Start-Schwellwert von 85% runter auf 50% und guck ob das Problem weiterhin existiert
- Wie swinstate bereits gesagt hat, nach jedem Neustart  prueft DPL ernuet den Start-Schwellwert. Ist er unter 85%, dann startet der Wechselrichter nicht mehr (ausser wenn Batterie Nachts sogar teilweise geladen nutzen eingeschaltet ist, was du nicht hast).

**Reply by Michel83** [reply]
Also zum Setup :
Victron 100-50, Victron Smart Shunt, 25.6v 100a lifepo4.

Da der Smart Shunt ja den SOC liefert dachte ich das die Spnnungseinstellungen hier keine Wirkung haben.

Neustarts konnte ich so keine Festellen, es ist ein ESP32 mit 4mb und USB - C Anschluss hängt an einen 5A Netzgerät.

Das mit der Batterienutzung habe ich gemacht und es lief die Nacht auch durch wie eingestellt bis auf 21%

Gestern mittag ist das gleiche passiert die Batterie war bei 90% und OpenDtu ging in den FSP Funktionierte ungefähr eine Stunde und auch hier ging der Wechselrichter aus, sieht man auch in dem Diagramm.

Was mir noch einfällt ich habe den alten HMS-450 ca. 15cm neben der DTU montiert, kann das veielicht ein Problem sein ?

**Comment by Manos1966** [comment]
> Das mit der Batterienutzung habe ich gemacht und es lief die Nacht auch durch wie eingestellt bis auf 21%

Sehr gut 👍 Also, es geht doch.

> Gestern mittag ...

brauchen wir nicht darueber diskutieren, wenn es naher (gestern Nacht) funktioniert hat...

> 25.6v 100a lifepo4.

Habe ich mir gedacht.... Die Batterie Spannungs Schwellwerte anpassen zu etwas realistisches. 

>  HMS-450 ca. 15cm neben der DTU montiert

machen die meisten von uns... und manchmal klappt es weniger gut. Du muss in die Console gucken, ob die OpenDTU viele Re-Transmits macht bis sie eine Antwort von der HMS-450 erfolgreich empfangen hat. Falls es viele sind, dann ist sie zu nah.
Es ist aber unwahrscheinlich, dass dies der Grund deines Problems ist.

Jetzt dass es scheint besser zu funktionieren, passe auch die Batterie Spannungs Schwellwerte an, und guck ob das Problem sich erledigt hat.

**Reply by Manos1966** [reply]
Dazu: Du sammelst sowieso Daten. Dann, verfolge wie oft deine OpenDTU neu startet.
Das kannst sehen wenn du die Speicher Auslastung verfolgst (beim jedem Neustart fängt sie von null an).

**Comment by Michel83** [comment]
Hi, also Reboot schließe ich aus. Er müsste doch dann auch den Betriebszeit Zähler zurücksetzten.
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/36561525/137afa8d-51a1-424c-804f-a42ff06cf958)

Habe die Spannungen jetzt mal so eingestellt:

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/36561525/bcca5756-7782-4117-b8ba-fa7605cf4e20)


bis jetzt läuft alles.

Allerdings hatte ich auch heute mal im Log vom DPL ein paar mal Waiting of Timesync, oder so ähnlich gelesen bevor er eingeschaltet hat.
Leider konnte ich den Log in dem Moment nicht aufzeichnen.

Ich denke ich beobachte das jetzt mal weiter.

Großen Dank schon mal.

**Comment by Oldie-driver** [comment]
Das hier hat mir sehr weiter geholfen, Danke!

Ich hatte die Startspannung als Wert interpretiert, ab dem der Inverter wieder eingeschaltet werden soll, nachdem der Stoppwert erreicht wurde. Ein Ausschalten passiert auch durch den mal auftretenden FB Überstrom. Nur softwaremäßig wieder einschalten klappte nicht lange, wenn die Spannung unter der Startspannung  aber über der Stopp Spannung war.

Dieses Problem habe ich vom Laderegler aus gelöst, wenn die Spannung  unter 21V ist, wird der Hoymiles auf DC Seite getrennt und erst ab 25V mit Einschaltstrombegrenzung ( Power MosFET mit Optokopler in der Ansteuerung ) wieder zugeschaltet. Der Startwert von OpenDTU daher jetzt auf 23V gestellt, Stopp auf 21,3V.  Somit kann Home Assistent den Hoymiles wieder meistens selbst starten wenn der Fehler 46 oder 47 wieder mal auftrat. Dabei kann ist der hier mit 11A noch im Limit, der Fehler kann eigentlich nicht am HM-400 auftreten.
